research/2022-07-15-relay-anonymity.md

```diff
@@ -0,0 +1,351 @@
+With this, the receiver discloses its participation in Waku on the gossipsub layer.
+We will leave the analysis of libp2p gossipsub to a future article within this series, and only provide a few hints and pointers here.
+
+Waku offers k-anonymity regarding content topic interest in the global adversary model.
+[K-anonymity](https://en.wikipedia.org/wiki/K-anonymity) in the context of Waku means an attacker can link receivers to content topics with a maximum certainty of $1/k$.
+The larger $k$, the less certainty the attacker gains.
+Receivers basically hide in a pool of $k$ content topics, any subset of which could be topics they subscribed to.
+The attacker does not know which of those the receiver actually subscribed to,
+and the receiver enjoys [plausible deniability](https://en.wikipedia.org/wiki/Plausible_deniability#Use_in_cryptography) regarding content topic subscription.
+Assuming there are $n$ Waku content topics, a receiver has $n$-anonymity with respect to association to a specific content topic.
+
+Technically, Waku allows distributing messages over several libp2p pubsub topics.
+This yields $k$-anonymity, assuming $k$ content topics share the same pubsub topic.
+However, if done wrongly, such sharding of pubsub topics can breach anonymity.
+A formal specification of anonymity-preserving topic sharding building on the concepts of [partitioned topics](https://specs.status.im/spec/10#partitioned-topic) is part of our roadmap.
+
+Also, Waku is not directly concerned with 1:1 communication, so for this post, 1:1 communication is out of scope.
+A 1:1 communication protocol might be added to Waku in the future.
+Similar to topic sharding, it would maintain receiver anonymity leveraging [partitioned topics](https://specs.status.im/spec/10#partitioned-topic).
+
+#### Sender Anonymity
+
+We define sender anonymity as *unlinkability of users' identities and the data they send and/or related actions*.
+Because the data in the context of Waku is Waku messages, sender anonymity corresponds to *sender-message unlinkability*.
+
+In summary, Waku offers sender anonymity in the single node and local attacker models because of [Waku's strict no sign policy](https://rfc.vac.dev/spec/11/#signature-policy),
+which has its origins in the [Ethereum consensus specs](https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/p2p-interface.md#why-are-we-using-the-strictnosign-signature-policy).
+[17/WAKU-RLN-RELAY](https://rfc.vac.dev/spec/17/) and [18/WAKU2-SWAP](https://rfc.vac.dev/spec/18/) mitigate replay and injection attacks.
+
+Waku currently does not offer sender anonymity in stronger attacker models.
+We will cover this in more detail in later sections.
+
+### Anonymity Trilemma
+
+[The Anonymity trilemma](https://freedom.cs.purdue.edu/projects/trilemma.html) states that only two out of *strong anonymity*, *low bandwidth*, and *low latency* can be guaranteed in the global on-net attacker model.
```

Comment:
it might make sense to include an image with illustration of this triangle/trilemma, this also breaks up the text mnd makes it easier to read

A 4th dimension here is also user distribution, so how many people are using the network. Stronger requirements means fewer people use it. This also relates to "k anonymity"

Reply:
Ill address the image later.

The 4th factor is added in 3e9c42f7